packages/widgetbook/lib/src/fields/field.dart

```diff
@@ -99,7 +99,7 @@ abstract class Field<T> {
     // value was null (i.e. had the nullability symbol).
     final rawNewValue = codec.toParam(value);
     final newValue =
-        isNull(groupMap)
+        value == null && isNull(groupMap)
             ? '${Field.nullabilitySymbol}$rawNewValue'
             : rawNewValue;
```

Comment:
As the comment mentions, it preserves the nullability when the `value` changes.
Since you want to always force field out of nullability when value changes, which I agree is better UX, then we should just remove the conditional update as follows:

```dart
void updateField(BuildContext context, String group, T value) {
  final state = WidgetbookState.of(context);
  final stringifiedValue = codec.toParam(value);

  state.updateQueryField(
    group: group,
    field: name,
    value: stringifiedValue,
  );
}
```

_(please make sure the test is in sync with the new implementation)_